quantum/dynamic_keymap.c

```diff
@@ -121,10 +121,12 @@ uint16_t dynamic_keymap_get_keycode(uint8_t layer, uint8_t row, uint8_t column)
 
 void dynamic_keymap_set_keycode(uint8_t layer, uint8_t row, uint8_t column, uint16_t keycode) {
     if (layer >= DYNAMIC_KEYMAP_LAYER_COUNT || row >= MATRIX_ROWS || column >= MATRIX_COLS) return;
-    void *address = dynamic_keymap_key_to_eeprom_address(layer, row, column);
+    uint16_t prev_keycode = dynamic_keymap_get_keycode(layer, row, column);
+    void *   address      = dynamic_keymap_key_to_eeprom_address(layer, row, column);
     // Big endian, so we can read/write EEPROM directly from host if we want
     eeprom_update_byte(address, (uint8_t)(keycode >> 8));
     eeprom_update_byte(address + 1, (uint8_t)(keycode & 0xFF));
+    unregister_code(prev_keycode);
```

Comment:
As discussed on discord, this wont correctly handle keycode that are non-basic, and changing it to `unregister_code16` would then only handle basic+modded. Using `clear_keyboard()` covers more scenarios but still does not cover any side effects of quantum keycodes.

Im also not a fan of coupling storage directly to its eventual use, wherever in the tmk action code. I agree updating `via.c` to be responsible would be error prone for other eventual uses of `dynamic_keymap_set_keycode`, but it still feels like the cleaner workaround till a better solution is implemented.

Reply:
A reason I handled things in here was to cover the scenario of:
1. key pressed
2. dynamic keymap set
3. dynamic keymap set (again)
4. key released

Is changing `unregister_code` to handle non-basic keycodes an option?
Or creating a new, all-encompassing function to handle unregistering any code?

Reply:
Also, when you suggest to have `via.c` be responsible, would having the same logic but instead immediately before and after the call to `dynamic_keymap_set_keycode` within the `id_dynamic_keymap_set_keycode` case suffice? Or would there be more to having this happen in `via.c`?

Reply:
resolving since I made `via.c` responsible to handle this

Reply:
Unresolving as its still using `unregister_code`.